content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+
+## Primary changes
+
+This section gives an overview of the main breaking changes in Prisma 5.
+
+### Removal of <inlinecode>rejectOnNotFound</inlinecode> parameter
+
+From Prisma version 5.0.0, the deprecated parameter `rejectOnNotFound` has been removed. Depending on if your project used `rejectOnNotFound` per query or globally, there will be be different ways of updating your code.
+
+If you are using the `rejectOnNotFound` parameter on a per-query basis, then follow our steps for [updating your code at the query level](#replacing-rejectonnotfound-enabled-at-the-query-level)
+
+If instead you have set up the `rejectOnNotFound` parameter at the client level, you will need to follow [the steps for updating your code at the client level](#replacing-rejectonnotfound-enabled-at-the-client-level)
+
+#### Replacing <inlinecode>rejectOnNotFound</inlinecode> enabled at the query level.
+
+If you enable `rejectOnNotFound` on a per-query basis, you will need to replace your usage at the query level.
+
+The following example:
+
+```js
+prisma.user.findFirst({
+  where: { name: 'Alice' },
+  rejectOnNotFound: true,
+})
+```
+
+Is converted to:
+
+```js
+prisma.user.findFirstOrThrow({
+  where: { name: 'Alice' },
+})
+```
+
+If you use a custom error handler like the following:
+
+```js
+prisma.user.findFirst({
+  where: { name: 'Alice' },
+  rejectOnNotFound: () => new UserNotFoundError(),
+})
+```
+
+You will need to modify your code to handle the errors thrown by `...OrThrow` methods.
+
+```js
+try {
+  prisma.user.findFirstOrThrow({
+    where: { name: 'Alice' },
+  })
+} catch (err) {
+  if (err.code === 'P2025') {
+    throw new UserNotFoundError()
+  }
+  throw err
+}
+```
+
+You could also create a reusable error adapter which can be used in a `catch` in your code.
+
+```js
+const adaptError = (customThrowFn) => (error) => {
+  if (error.code === 'P2025') {
+    throw customThrowFn()
+  }
+  throw error
+}
+
+const user = await prisma.user.findFirstOrThrow({
+  where: { name: 'Alice' },
+}).catch(adaptError(() => new MyCustomError())
+```
+
+#### Replacing <inlinecode>rejectOnNotFound</inlinecode> enabled at the client level.
+
+For cases like these below:
+
+```js
+// Example 1
+const prisma = new PrismaClient({
+  rejectOnNotFound: true,
+})
+
+// Example 2
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findUnique: true,
+  },
+})
+```
+
+You will need to update your codebase to use `findUniqueOrThrow` and `findFirstOrThrow` instead of `findUnique` and `findFirst`, depending on which calls you would like to throw.
+
+If instead you use a custom error handler with the `rejectOnNotFound` property, like these examples:
+
+```js
+// Example 3
+const prisma = new PrismaClient({
```

Comment:
I'd also add example migration for these 2 cases, they are simpler than for Example 5

Reply:
Cna you suggest what exactly needs to be change @SevInf?
Any incomplete input is better for @jharrell than none and having to figure it out.

Reply:
@SevInf resolving this for now, but if you have suggestions I'm happy to add them in a post-launch update.